tests/device/test_http_server/test_http_server.ino

```diff
@@ -26,21 +27,27 @@ void setup()
     BS_RUN(Serial);
 }
 
+void handle_request()
+{
+    for (uint8_t i=0; i<server.args(); i++){
+        // skip "plain" which is automatically added during arg parsing for post's
+        if (server.argName(i) == "plain")
+            continue;
+        if(i > 0)
+            siteData += "\n";
+        siteData += server.argName(i) + " = " + server.arg(i);
+    }
+    siteHits++;
+    server.send(200, "text/plain", siteData);
+}
+
 
 TEST_CASE("HTTP GET Parameters", "[HTTPServer]")
 {
     {
         siteHits = 0;
-        server.on("/get", HTTP_GET, [](){
-            siteData = "";
-            for (uint8_t i=0; i<server.args(); i++){
-                if(i > 0)
-                    siteData += "\n";
-                siteData += server.argName(i) + " = " + server.arg(i);
-            }
-            siteHits++;
-            server.send(200, "text/plain", siteData);
-        });
+        siteData = "";
```

Comment:
I suspect this code will work anyways, but to be equivalent to the original code, this line should go in the handle_request() function.

Reply:
That would work for all but the last test, "HTTP Upload", where the upload code, which sets ```siteData``` is called before ```handle_request```.

Reply:
@devyte To clarify - setting ```siteData``` in ```handle_request``` would break the "HTTP Upload" test.  I found this empirically as my initial implementation was as you had suggested and that's where I found that the upload callback was called before the request callback.

Reply:
@liebman thought I had answered this sorry.
You're right, I didn't realize the last test was different.